lib/ansible/modules/files/archive.py

```diff
@@ -346,153 +365,176 @@ def main():
+                                                arcfile.write(b_fullpath, b_arcname)
                                             else:
-                                                arcfile.add(fullpath, arcname, recursive=False)
+                                                arcfile.add(b_fullpath, b_arcname, recursive=False)
 
-                                            successes.append(fullpath)
+                                            b_successes.append(b_fullpath)
                                         except Exception as e:
-                                            errors.append('Adding %s: %s' % (path, to_native(e)))
+                                            errors.append('Adding %s: %s' % (to_native(b_path), to_native(e)))
                         else:
-                            if format == 'zip':
-                                arcfile.write(path, match_root.sub('', path))
+                            path = to_native(b_path, errors='surrogate_or_strict')
+                            arcname = to_native(b_match_root.sub(b'', b_path), errors='surrogate_or_strict')
+                            if fmt == 'zip':
+                                arcfile.write(path, arcname)
                             else:
-                                arcfile.add(path, match_root.sub('', path), recursive=False)
+                                arcfile.add(path, arcname, recursive=False)
 
-                            successes.append(path)
+                            b_successes.append(b_path)
 
                 except Exception as e:
-                    module.fail_json(msg='Error when writing %s archive at %s: %s' % (format == 'zip' and 'zip' or ('tar.' + format), dest, to_native(e)),
-                                     exception=format_exc())
+                    expanded_fmt = 'zip' if fmt == 'zip' else ('tar.' + fmt)
+                    module.fail_json(
+                        msg='Error when writing %s archive at %s: %s' % (expanded_fmt, dest, to_native(e)),
+                        exception=format_exc()
+                    )
 
                 if arcfile:
                     arcfile.close()
                     state = 'archive'
 
-                if format == 'xz':
-                    with lzma.open(dest, 'wb') as f:
+                if fmt == 'xz':
+                    with lzma.open(b_dest, 'wb') as f:
                         f.write(arcfileIO.getvalue())
                     arcfileIO.close()
 
                 if errors:
                     module.fail_json(msg='Errors when writing archive at %s: %s' % (dest, '; '.join(errors)))
 
         if state in ['archive', 'incomplete'] and remove:
-            for path in successes:
+            for b_path in b_successes:
                 try:
-                    if os.path.isdir(path):
-                        shutil.rmtree(path)
+                    if os.path.isdir(b_path):
+                        shutil.rmtree(v_path)
```

Comment:
Probably a typo here?
```suggestion
                        shutil.rmtree(b_path)
```

Reply:
Indeed.